src/it/scala/com/exasol/cloudetl/avro/AvroDataImporterIT.scala

```diff
@@ -0,0 +1,354 @@
+      .assertResultSet(
+        table()
+          .row(expectedTimestamp1)
+          .row(expectedTimestamp2)
+          .matches()
+      )
+  }
+
+  test("imports float") {
+    val EPS = java.math.BigDecimal.valueOf(0.0001)
+    AvroChecker(getBasicSchema("\"float\""), "FLOAT", "float")
+      .withInputValues(List(3.14f, 2.71f))
+      .assertResultSet(
+        table()
+          .row(CellMatcherFactory.cellMatcher(3.14, STRICT, EPS))
+          .row(CellMatcherFactory.cellMatcher(2.71, STRICT, EPS))
+          .matches()
+      )
+  }
+
+  test("imports double") {
+    AvroChecker(getBasicSchema("\"double\""), "DOUBLE", "double")
+      .withInputValues(List(3.14, 2.71))
+      .assertResultSet(
+        table()
+          .row(java.lang.Double.valueOf(3.14))
+          .row(java.lang.Double.valueOf(2.71))
+          .matches()
+      )
+  }
+
+  test("imports bytes") {
+    AvroChecker(getBasicSchema("\"bytes\""), "VARCHAR(20)", "bytes")
+      .withInputValues(List(ByteBuffer.wrap("hello".getBytes(UTF_8))))
+      .assertResultSet(
+        table()
+          .row("hello")
+          .matches()
+      )
+  }
+
+  test("imports bytes (decimal)") {
+    val decimal1 = ByteBuffer.wrap(new BigDecimal("123456").unscaledValue().toByteArray())
+    val decimal2 = ByteBuffer.wrap(new BigDecimal("12345678").unscaledValue().toByteArray())
+    val inner = """{"type":"bytes","logicalType":"decimal","precision":8,"scale":3}"""
+    AvroChecker(getBasicSchema(inner), "DECIMAL(8,3)", "bytes_decimal")
+      .withInputValues(List(decimal1, decimal2))
+      .assertResultSet(
+        table()
+          .row(java.lang.Double.valueOf(123.456))
+          .row(java.lang.Double.valueOf(12345.678))
+          .matches(NO_JAVA_TYPE_CHECK)
+      )
+  }
+
+  test("imports fixed") {
+    val schema = getBasicSchema("""{"type":"fixed","name":"fixed", "size":5}""")
+    val fixedSchema = new Schema.Parser().parse(schema).getField("column").schema()
+    val fixedData = new GenericData.Fixed(fixedSchema)
+    fixedData.bytes("fixed".getBytes(UTF_8))
+    AvroChecker(schema, "VARCHAR(20)", "fixed")
+      .withInputValues(List(fixedData))
+      .assertResultSet(
+        table()
+          .row("fixed")
+          .matches()
+      )
+  }
+
+  test("imports fixed (decimal)") {
+    val inner = """{"type":"fixed","name":"fx","size":7,"logicalType":"decimal","precision":7,"scale":4}"""
+    val schema = getBasicSchema(inner)
+    val fixedSchema = new Schema.Parser().parse(schema).getField("column").schema()
+    val fixedData = new Conversions.DecimalConversion().toFixed(
+      new BigDecimal("0.0123"),
+      fixedSchema,
+      LogicalTypes.decimal(7, 4)
+    )
+    AvroChecker(schema, "DECIMAL(7,4)", "fixed_decimal")
+      .withInputValues(List(fixedData))
+      .assertResultSet(
+        table()
+          .row(java.lang.Double.valueOf(0.0123))
+          .matches(NO_JAVA_TYPE_CHECK)
+      )
+  }
+
+  test("imports string") {
```

Comment:
Are nulls supported in Avro?

Reply:
Yes, it does. But to use null the Avro schema should be `union` type. There is a test down with union for it.

Reply:
Yes, it does. But to use null the Avro schema should be `union` type. There is a test down with union for it.